src/htmx.js

```diff
@@ -1906,32 +1906,45 @@ return (function () {
             }
         }
 
+        // Handle submit buttons/inputs that have the form attribute set
+        // see https://developer.mozilla.org/docs/Web/HTML/Element/button
+         function maybeSetLastButtonClicked(evt) {
+            var elt = closest(evt.target, "button, input[type='submit']");
```

Comment:
feel like the first bit of these two functions could be refactored to use a common method, maybe `findRelatedForm(elt)` that returns nothing if there is none, then the logic can be boiled down to a check and set in both.

does that make sense?